autoload/fern/fri.vim

```diff
@@ -131,14 +131,11 @@ function! s:encode(str, pattern) abort
 endfunction
 
 function! s:decode(str) abort
-  let str = a:str
-  let [hex, s, e] = matchstrpos(str, '%\zs[0-9a-fA-F]\{2}')
-  while !empty(hex)
-    let repl = nr2char(str2nr(hex, 16))
-    let str = substitute(str, '%' . hex, escape(repl, '&'), 'ig')
-    let [hex, s, e] = matchstrpos(str, '%\zs[0-9a-fA-F]\{2}', s + 1)
-  endwhile
-  return str
+  if type(a:str) !=# v:t_string
+    return a:str
+  endif
```

Comment:
I don't think explicit check here is required while the decode only accepts a string value thus passing a non-string value is the wrong usage. While Vim script is slow, I usually avoid such condition checks for developers.

Or am I missing some cases?

Reply:
Flag type query (eg. `-drawer`) parsed to `v:true` and decoding it.
https://github.com/lambdalisue/fern.vim/pull/179/files/86eb1573b1838dcd202d3a096f2e300296392001#diff-7eb995c2fea3b2735ea69941139497efR74-R77

If remove this, so test failed.

Reply:
I see. Then the usage of `s:decode()` in `s:parse_query()` was wrong. So please fix it by

1. Remove that check in `s:decode()`
2. Add type check on `s:parse_query()` like below

```vim
        \   s:decode(v[0]): type(v[1]) is# v:t_string ? s:decode(v[1]) : v[1],
```

Reply:
`s:parse_query()` calls `map()` twice.
I want to fix this so that it is called only once. Is it OK?